draco/analysis/delay.py

```diff
@@ -979,29 +985,34 @@ class DelayPowerSpectrumGeneralEstimator(
     """Class to measure delay power spectrum of general container via Gibbs sampling."""
 
 
-class DelaySpectrumWienerEstimator(DelayGeneralContainerBase):
-    """Class to measure delay spectrum of general container via Wiener filtering.
+class DelayTransformFFT(DelayGeneralContainerBase, DelayGibbsSamplerBase):
+    """Class to measure the delay spectrum of a general container via ifft.
 
-    The spectrum is calculated by applying a Wiener filter to the input frequency
-    spectrum, assuming an input model for the delay power spectrum of the signal and
-    that the noise power is described by the weights of the input container. See
-    https://arxiv.org/abs/2202.01242, Eq. A6 for details.
+    This class assumes that frequency spectrum gaps have been dealt with
+    in some way, so zero-weight frequencies are ignored.
     """
 
-    def setup(self, dps=None):
-        """Set the delay power spectrum to use as the signal covariance.
+    powerspectrum = config.Property(proptype=bool, default=False)
 
-        Parameters
-        ----------
-        dps : `containers.DelaySpectrum`
-            Delay power spectrum for signal part of Wiener filter.
-        """
-        self.dps = dps
+    def setup(self):
+        """Stop any frequency cut from being applied."""
+        self.log.info(
+            "Setting `freq_frac = -1.0` for FFT. "
+            "This means that _all_ frequencies will be "
+            "included in the delay transform, includeing "
+            "those which are masked."
+        )
+        self.freq_frac = -1.0
+
+        if self.powerspectrum:
+            # TODO: this seems to work, but I'm not sure if
+            # it's the correct way to do this
+            self._create_output = DelayGibbsSamplerBase()._create_output
```

Comment:
I don't think it's _incorrect_, per se, but I think it might cause confusion, in that people trying to debug code later might not realise that the `_create_output` defined just below might be completely skipped in the program flow unexpectedly.

I don't know if there are implications for delaying this check, but, could you just have:
```
if self.powerspectrum:
  return DelayGibbsSamplerBase()._create_output(self, ss, delays, coord_axes)
```
at the top of the `_create_output` defined below?

It would also save you from having to repeat this diversion in `DelaySpectrumWienerEstimator.setup`.

Reply:
I agree that it's confusing, and I think your suggestion would be the right way to go about it. However, this line ends up getting somewhat refactored in #321 (the `_create_output` methods are defined independently of other functionality), so I think that PR is where we should consider changes.

Reply:
That's fair.